python/grass/script/tests/grass_script_setup_test.py

```diff
@@ -30,40 +28,91 @@ def test_init_as_context_manager(tmp_path):
 
 
-def test_init_finish_global_functions(tmp_path):
+def test_init_finish_global_functions_with_env(tmp_path):
     """Check that init and finish global functions work"""
     location = "test"
     gs.core._create_location_xy(tmp_path, location)  # pylint: disable=protected-access
-    gs.setup.init(tmp_path / location)
-    gs.run_command("g.region", flags="p")
-    session_file = os.environ["GISRC"]
-    gs.setup.finish()
+    env = os.environ.copy()
+    gs.setup.init(tmp_path / location, env=env)
+    gs.run_command("g.region", flags="p", env=env)
+    session_file = env["GISRC"]
+    gs.setup.finish(env=env)
 
     assert not os.path.exists(session_file)
 
 
+def test_init_finish_global_functions_capture_strerr0(tmp_path):
+    """Check that init and finish global functions work with global env"""
+
+    def init_finish(queue):
+        gs.set_capture_stderr(True)
+        location = "test"
+        gs.core._create_location_xy(  # pylint: disable=protected-access
+            tmp_path, location
+        )
+        gs.setup.init(tmp_path / location)
+        gs.run_command("g.region", flags="p")
+        runtime_present = bool(os.environ.get("GISBASE"))
+        queue.put((os.environ["GISRC"], runtime_present))
+        gs.setup.finish()
+
+    session_file, runtime_present = run_in_subprocess(init_finish)
+    assert session_file, "Expected file name from the subprocess"
+    assert runtime_present, "Runtime (GISBASE) should be present"
+    assert not os.path.exists(session_file), "Session file not deleted"
+
+
+def test_init_finish_global_functions_capture_strerrX(tmp_path):
+    """Check that init and finish global functions work with global env"""
+
+    def init_finish(queue):
+        gs.set_capture_stderr(True)
+        location = "test"
+        gs.core._create_location_xy(  # pylint: disable=protected-access
+            tmp_path, location
+        )
+        gs.setup.init(tmp_path / location)
+        gs.run_command("g.region", flags="p")
+        runtime_present = bool(os.environ.get("GISBASE"))
+        session_file = os.environ["GISRC"]
+        gs.setup.finish()
+        runtime_present_after = bool(os.environ.get("GISBASE"))
+        queue.put((session_file, runtime_present, runtime_present_after))
+
+    session_file, runtime_present, runtime_present_after = run_in_subprocess(
+        init_finish
+    )
+    assert session_file, "Expected file name from the subprocess"
+    assert runtime_present, "Runtime (GISBASE) should be present"
+    assert not os.path.exists(session_file), "Session file not deleted"
+    # This is testing the current implementation behavior, but it is not required
+    # to be this way in terms of design.
+    assert runtime_present_after, "Runtime should continue to be present"
+
+
 def test_init_finish_global_functions_capture_strerr(tmp_path):
-    """Check that init and finish global functions work"""
+    """Check that init and finish global functions work with global env"""
```

Comment:
This test still repeatedly times out after 5 minutes (300 secs), even after a rerun, where it times out for all three configurations

Reply:
The time out is a consequence of a failing subprocess and then waiting for the result from it...there is probably a better way how to handle that situation.

Anyway, the actual error is there too:

```
Process Process-21:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/runner/work/grass/grass/python/grass/script/tests/grass_script_setup_test.py", line 128, in init_finish
    queue.put((os.environ["GISRC"], runtime_present_during, runtime_present_after))
               ~~~~~~~~~~^^^^^^^^^
  File "<frozen os>", line 685, in __getitem__
KeyError: 'GISRC'
```

The error was not there before in the previous commit (before the merge). I need to check locally.

Reply:
Did you check that in the end?

Reply:
Just fixed that now!